youtube_dl/extractor/pandoratv.py

```diff
@@ -0,0 +1,40 @@
+# encoding: utf-8
+import re
+import json
+
+from .common import InfoExtractor
+from ..compat import (
+    compat_parse_qs,
+    compat_urllib_parse_urlparse,
+)
+from ..utils import (
+    find_xpath_attr,
+    unified_strdate,
+    get_element_by_attribute,
+    int_or_none,
+    qualities,
+)
+
+
+class PandoraTVIE(InfoExtractor):
+    _VALID_URL = r'http://channel.pandora.tv/channel/video.ptv\?ch_userid=(?P<userid>.*?)&prgid=(?P<prgid>.*?)&.*'
```

Comment:
Subdomains are not matched: http://jp.channel.pandora.tv/channel/video.ptv?c1=&prgid=53294230&ch_userid=mikakim&ref=main&lot=cate_01_2.